internal/server/http/server_test.go

```diff
@@ -126,24 +126,55 @@ func TestBuildProxyMode(t *testing.T) {
 		})
 	}
 }
+func TestBuildSecureMode(t *testing.T) {
+	// proxyServer := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	// 	io.WriteString(w, "Proxied")
```

Comment:
the problem in the tests seems to be here, you return "Proxied" but expected `body:   "Handled"` line: 153

Reply:
Yeah, but if I use proxy mode `killgrave.ProxyNone` it wouldn't through the proxy but through the mock. 

I want to reproduce the example as in previous test (TestBuildProxyMode), `ProxyNone_Hit` case, but in this case with HTTPS. 

In HTTP case
```
"ProxyNone_Hit": {
	mode:   killgrave.ProxyNone,
	url:    "/testRequest",
	body:   "Handled",
	status: http.StatusOK,
},
```
 it returns `Handled`, but not in here 🤔